src/aggregations/bucket-aggregations/auto-histogram-aggregation-base.js

```diff
@@ -0,0 +1,201 @@
+'use strict';
+
+const has = require('lodash.has');
+const isNil = require('lodash.isnil');
+
+const {
+    util: { invalidParam }
+} = require('../../core');
+
+const BucketAggregationBase = require('./bucket-aggregation-base');
+
+const invalidDirectionParam = invalidParam('', 'direction', "'asc' or 'desc'");
+
+/**
+ * The `AutoHistogramAggregationBase` provides support for common options used across
+ * various auto-histogram `Aggregation` implementations like
+ * Auto Date Histogram aggregation.
+ *
+ * **NOTE:** Instantiating this directly should not be required.
+ * However, if you wish to add a custom implementation for whatever reason,
+ * this class could be extended.
+ *
+ * @param {string} name The name which will be used to refer to this aggregation.
+ * @param {string} aggType Type of aggregation
+ * @param {string=} field The field to aggregate on
+ * @param {number=} bucket count to generate histogram over.
+ *
+ * @extends BucketAggregationBase
+ */
+class AutoHistogramAggregationBase extends BucketAggregationBase {
+    // eslint-disable-next-line require-jsdoc
+    constructor(name, aggType, field, buckets) {
+        super(name, aggType, field);
+
+        if (!isNil(buckets)) this._aggsDef.buckets = buckets;
+    }
+
+    /**
+     * Sets the histogram bucket count. Buckets are generated based on this value.
+     *
+     * @param {number=} bucket count to generate histogram over.
```

Comment:
```suggestion
     * @param {number} bucket count to generate histogram over.
```

The parameter is not optional.